src/hdkeys.hpp

```diff
@@ -97,6 +97,7 @@ class HDKeys {
         Util::SecFree(ikmHkdf);
         Util::SecFree(skBn);
         Util::SecFree(okmHkdf);
+        Util::SecFree(skBytes);
```

Comment:
this leak could possibly cause the exhaustion of libsodium allocations. But the security of this doesn't seem all the way thought through. Apart from the fact that allocating the `bn_t` object itself in secure memory, the private key is then copied to the stack.